File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
def date_to_days(date_str):
    y, m, d = map(int, date_str.split('.'))
    return (y * 12 * 28) + (m * 28) + d

def solution(today, terms, privacies):
    answer = []
    
    today_days = date_to_days(today)
    
    terms_dict = {}
    for term in terms:
        t_type, t_month = term.split()
        terms_dict[t_type] = int(t_month) * 28
        
    for idx, privacy in enumerate(privacies):
        date, t_type = privacy.split()
        
        privacy_days = date_to_days(date)

        expire_days = privacy_days + terms_dict[t_type]

        if expire_days <= today_days:
            answer.append(idx + 1)
            
    return answer

File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(const char* today, const char* terms[], size_t terms_len, const char* privacies[], size_t privacies_len) {
    int* answer = (int*)malloc(sizeof(int) * privacies_len);
    int ans_idx = 0;
    
    int ty, tm, td;
    sscanf(today, "%d.%d.%d", &ty, &tm, &td);
    int today_days = (ty * 12 * 28) + (tm * 28) + td;
    
    int term_map[26] = {0};
    for (size_t i = 0; i < terms_len; i++) {
        char type;
        int months;
        sscanf(terms[i], "%c %d", &type, &months);
        term_map[type - 'A'] = months * 28; 
    }
    
    for (size_t i = 0; i < privacies_len; i++) {
        int py, pm, pd;
        char type;
        
        sscanf(privacies[i], "%d.%d.%d %c", &py, &pm, &pd, &type);
        int privacy_days = (py * 12 * 28) + (pm * 28) + pd;
        
        if (privacy_days + term_map[type - 'A'] <= today_days) {
            answer[ans_idx++] = i + 1;
        }
    }
    
    return answer;
}